cmd/entrypoint/secrets.go

```diff
@@ -313,6 +329,43 @@ func ReconcileSecrets(ctx context.Context, sh *SnapshotHolder) error {
 	return nil
 }
 
+// Returns secretName, secretNamespace from a provided (unstructured) filter if it contains a secret
+// Returns empty strings when the secret name and/or namespace could not be found
+func findFilterSecret(filter *unstructured.Unstructured) (string, string) {
+	// Just making extra sure this is actually a Filter
+	if filter.GetKind() != "Filter" && filter.GetKind() != "Filters" {
+		return "", ""
+	}
+	// Only Oauth2 Filters have secrets, although they dont need to have them.
+	// Yes this is disgusting. It is what it is...
+	filterContents := filter.UnstructuredContent()
+	filterSpec := filterContents["spec"]
+	if filterSpec != nil {
+		oauthFilter := filterSpec.(map[string]interface{})["OAuth2"]
```

Comment:
my only question here, is there a risk that we panic due to type assertion failing? Do we have guarantees that the unstructured content within the Spec field is already valid "yaml" thus ensuring this doesn't panic.

I don't have an answer but just want to make sure we think about that.